src/cli.rs

```diff
@@ -10,6 +14,95 @@ pub struct Args {
 
 #[derive(Subcommand)]
 pub enum SubCommands {
-    Client,
-    Server,
+    Install {
+        #[command(subcommand)]
+        subcommand: InstallSubcommands
+    }
 }
+
+#[derive(Subcommand)]
+pub enum InstallSubcommands {
+    Client {
+        #[arg(short='M',long)]
+        minecraft_version: Option<String>,
+
+        #[arg(short='L',long)]
+        loader_version: Option<String>,
+        #[arg(short='s',long)]
+        snapshot: bool,
+        #[arg(short='b',long)]
+        loader_beta: bool,
+
+        #[arg(short, long)]
+        no_profile: bool,
+        #[arg(short, long, value_name = "'.minecraft' DIRECTORY")]
+        install_dir: Option<PathBuf>
+    },
+    Server {
+        #[arg(short='M',long)]
+        minecraft_version: Option<String>,
+        #[arg(short='L',long)]
+        loader_version: Option<String>,
+
+        #[arg(short, long)]
+        create_scripts: bool,
+        #[arg(short, long)]
+        download_server: bool,
+        #[arg(short, long)]
+        install_dir: PathBuf
+    }
+}
+
+
+pub async fn cli(args: SubCommands) -> anyhow::Result<()> {
+    match args {
+            SubCommands::Install { subcommand} => {
+                match subcommand {
+                    InstallSubcommands::Client { minecraft_version, loader_version, snapshot, loader_beta, no_profile, install_dir } => {
+                        let (mc_version_to_install, loader_version_to_install) =  get_versions(minecraft_version, loader_version, snapshot, loader_beta).await?;
+                        let install_dir = install_dir.unwrap_or_else(installer::get_default_client_directory);
+
+                            installer::install_client(ClientInstallation {
+                                minecraft_version: mc_version_to_install,
+                                loader_version: loader_version_to_install,
+                                install_dir,
+                                generate_profile: !no_profile}).await?;
+
+                    }
+                    InstallSubcommands::Server { .. } => {}
```

Comment:
Since you went to the effort of showing errors, this should either error or call `install_server`, which errors anyway.

Reply:
I am being told that server installs are coming in another PR. so I will ignore this for now.